scripts/setup_osx.sh

```diff
@@ -1,38 +1,248 @@
+    "require exactly 1 (namely: $_required_args_string), but got only" \
+    "${#_positionals[@]}.")"
+  test ${#_positionals[@]} -ge 1 || _PRINT_HELP=yes die "${error_msg}" 1
+  error_msg="$(echo "FATAL ERROR: There were spurious positional arguments" \
+    "--- we expect exactly 1 (namely: $_required_args_string), but got" \
+    "${#_positionals[@]} (the last one was: '${_positionals[*]: -1}').")"
+  test ${#_positionals[@]} -le 1 || _PRINT_HELP=yes die "${error_msg}" 1
+}
+
+assign_positional_args ()
+{
+  _positional_names=('_arg_mjkey_path' )
+
+  for (( ii = 0; ii < ${#_positionals[@]}; ii++))
+  do
+    error_msg="$(echo "Error during argument parsing, possibly an Argbash" \
+      "bug.")"
+    eval "${_positional_names[ii]}=\${_positionals[ii]}" || \
+      die "${error_msg}" 1
+  done
+}
+
+parse_commandline "$@"
+handle_passed_args_count
+assign_positional_args
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+
+# Utility functions
+print_error() {
+  printf "\e[0;31m%s\e[0m" "${1}"
+}
+
+print_warning() {
+  printf "\e[0;33m%s\e[0m" "${1}"
+}
+
+# Verify there's a file in the mjkey path
+test -f "${_arg_mjkey_path}" || _PRINT_HELP=yes die \
+  "The path ${_arg_mjkey_path} of the MuJoCo key is not valid." 1
+
+# Make sure that we're under the garage directory
+GARAGE_DIR="$(readlink -f $(dirname $0)/..)"
+cd "${GARAGE_DIR}"
+
+# File where environment variables are stored
+BASH_PROF="${HOME}/.bash_profile"
+
+# Install dependencies
+echo "Installing garage dependencies"
+echo "You will probably be asked for your sudo password"
+
+# Homebrew is required first to install the other dependencies
+hash brew 2>/dev/null || {
+  # Install the Xcode Command Line Tools
+  xcode-select --install
+  # Install Homebrew
+  /usr/bin/ruby -e "$(curl -fsSL \
+    https://raw.githubusercontent.com/Homebrew/install/master/install)"
+}
+
+# For installing garage: wget, bzip2, unzip, git
+# For building glfw: cmake
+# Required for OpenAI gym: cmake boost boost-python ffmpeg sdl2 swig wget
+# Required for OpenAI baselines: cmake openmpi
+brew update
+brew install \
+  wget \
+  bzip2 \
+  unzip \
+  git \
+  cmake \
+  boost \
+  boost-python \
+  ffmpeg \
+  sdl2 \
+  swig \
+  openmpi
+
+# Build GLFW because the brew version is too old
+GLFW_DIR="$(mktemp -d)/glfw"
+git clone https://github.com/glfw/glfw.git "${GLFW_DIR}"
+cd "${GLFW_DIR}"
+git checkout 0be4f3f75aebd9d24583ee86590a38e741db0904
+mkdir glfw-build
+cd glfw-build
+cmake -DBUILD_SHARED_LIBS=ON -DGLFW_BUILD_EXAMPLES=OFF \
+  -DGLFW_BUILD_TESTS=OFF -DGLFW_BUILD_DOCS=OFF ..
+make -j"$(nproc)"
+make install
+cd "${GARAGE_DIR}"
+
+# Leave a note in ~/.bashrc for the added environment variables
+if [[ "${_arg_set_envvar}" = on ]]; then
+  echo -e "\n# Added by the garage installer" >> "${BASH_PROF}"
 fi
 
-echo "Creating conda environment..."
+# Set up MuJoCo
+if [[ ! -d "${HOME}"/.mujoco/mjpro150 ]]; then
+  mkdir "${HOME}"/.mujoco
+  MUJOCO_ZIP="$(mktemp --suffix=_mujoco.zip)"
```

Comment:
I don't think OS X has --suffix option for mktemp.
```
$ mktemp --suffix=_mujoco.zip
mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix
```

Reply:
A solution could be to use the dry-run option to only output the name of the temporary file (not create it), and append the suffix.
This is a little bit unsafe if the file happens to exist, but since these file are the tmp folder and with random characters in their name, I don't think there's a chance this will happen.

Reply:
Just don't add a suffix or put it in a directory instead.